src/dualscreen.tsx

```diff
@@ -371,31 +441,66 @@ export class DualScreen extends KalturaPlayer.core.BasePlugin implements IEngine
   private _onActiveSlideChanged = (slideItem: SlideItem | null) => {
     if (!slideItem || slideItem.errored) {
       // deactivate dual-screen layout
-      this._setSingleLayout();
+      this._switchToHidden();
       return;
     }
     if (slideItem.portrait !== this._pipPortraitMode) {
       this._pipPortraitMode = slideItem.portrait;
-      if (this._layout === Layout.PIP && !this._inverse) {
+      if (this._layout === Layout.PIP) {
         // update PIP component
         this._setMode();
         return;
       }
     }
     // apply dual-screen layout
-    if (this._singleLayout) {
+    if (this._layout === Layout.Hidden) {
+      this._setDefaultMode();
       this._setMode();
     }
   };
 
   private _getThumbs() {
-    const kalturaCuePointService :any = this._player.getService('kalturaCuepoints');
-    kalturaCuePointService?.registerTypes([kalturaCuePointService.CuepointType.SLIDE]);
+    const kalturaCuePointService: any = this._player.getService('kalturaCuepoints');
+    kalturaCuePointService?.registerTypes([kalturaCuePointService.CuepointType.SLIDE, kalturaCuePointService.CuepointType.VIEW_CHANGE]);
   }
 
+  private _onSlideViewChanged = ({playerViewModeId}: ViewChangeData, viewModeLockState: boolean) => {
+    if (viewModeLockState) {
+      this._switchToHidden();
+      return;
+    }
+    switch (playerViewModeId) {
+      case 'parent-only':
+        if (this._layout !== Layout.SingleMedia) {
+          this._switchToSingleMedia();
+        }
+        break;
+      case 'pip-parent-in-large':
+        if (this._layout !== Layout.PIP) {
+          this._switchToPIP();
+        }
+        break;
+      case 'pip-parent-in-small':
+        if (this._layout !== Layout.PIPInverse) {
+          this._switchToPIPInverse();
+        }
+        break;
+      case 'sbs-parent-in-left':
+        if (this._layout !== Layout.SideBySide) {
+          this._switchToSideBySide();
+        }
+        break;
+      case 'sbs-parent-in-right':
+        if (this._layout !== Layout.SideBySideInverse) {
+          this._switchToSideBySideInverse();
+        }
+        break;
+    }
+  };
+
   private _getSecondaryMedia() {
     this._player.provider
-      .doRequest([{loader: SecondaryMediaLoader, params: {parentEntryId: this._player.getMediaInfo().entryId}}])
+      .doRequest([{loader: SecondaryMediaLoader, params: {parentEntryId: this._player.config.sources.id}}])
```

Comment:
Please fix like Oren said in the other PR

Reply:
good catch!
fixed